recipes/cgal/all/conanfile.py

```diff
@@ -14,63 +17,162 @@ class CgalConan(ConanFile):
     description = "C++ library that provides easy access to efficient and reliable algorithms"\
                   " in computational geometry."
     topics = ("cgal", "geometry", "algorithms")
+    package_type = "header-library"
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake"
-    exports_sources = "CMakeLists.txt"
+    generators = "CMakeDeps"
     short_paths = True
 
-    _cmake = None
-
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _min_cppstd(self):
+        return "14"
 
     @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "msvc": "191",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
 
-    def _configure_cmake(self):
-        if not self._cmake:
-            self._cmake = CMake(self)
-            self._cmake.definitions["CGAL_HEADER_ONLY"] = "TRUE"
-            self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+    def export_sources(self):
+        export_conandata_patches(self)
 
-    def _patch_sources(self):
-        if scm.Version(self.version) < "5.3":
-            files.replace_in_file(self, os.path.join(self._source_subfolder, "CMakeLists.txt"),
-                                "CMAKE_SOURCE_DIR", "CMAKE_CURRENT_SOURCE_DIR")
-        else:
-            files.replace_in_file(self, os.path.join(self._source_subfolder, "CMakeLists.txt"),
-                                "if(NOT PROJECT_NAME)", "if(TRUE)")
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("boost/1.75.0")
-        self.requires("eigen/3.3.9")
+        self.requires("boost/1.78.0")
```

Comment:
Is it possible to use Boost 1.82.0 instead? 
```suggestion
        self.requires("boost/1.82.0")
```